apps/nacl_demos/nacl_demos.cpp

```diff
@@ -70,15 +70,18 @@ void completion_callback(void *data, int32_t flags) {
 extern "C" void *halide_malloc(void *, size_t);
 extern "C" void halide_free(void *, void *);
 
-buffer_t ImageToBuffer(const ImageData &im) {
-    buffer_t buf;
-    memset(&buf, 0, sizeof(buffer_t));
+halide_nd_buffer_t<2> ImageToBuffer(const ImageData &im) {
+    halide_nd_buffer_t<2> buf;
     buf.host = (uint8_t *)im.data();
-    buf.extent[0] = im.size().width();
-    buf.stride[0] = 1;
-    buf.extent[1] = im.size().height();
-    buf.stride[1] = im.stride()/4;
-    buf.elem_size = 4;
+    //buf.dim[0] = halide_dimension_t(0, im.size().width(), 1);
```

Comment:
Should either use this commented code (preferred... looks more concise than the alternative) or delete it.